src/core/execute.c

```diff
@@ -2958,6 +2958,23 @@ static int exec_parameters_get_cgroup_path(const ExecParameters *params, char **
         return using_subcgroup;
 }
 
+static int exec_context_cpu_affinity_from_numa(const ExecContext *c, CPUSet *ret) {
+        assert(c);
+        assert(ret);
+
+        if (!c->derive_cpu_affinity_from_numa_mask)
+                return 0;
```

Comment:
hmm, this check is pretty much unnecessary, no? the only caller checks this anyway... And semantically it might even be useful to convert the numa mask to cpu affinity if the bool isn't set...